sdks/java/core/build.gradle

```diff
@@ -69,6 +69,13 @@ test {
   }
 }
 
+// Avoid paranamer 2.7 from leaking into the user classpath. Paranamer is an
+// Avro transitive dependency that can produce runtime conflicts between
+// versions of Spark/Avro
+configurations.shadow {
```

Comment:
I think there's a better way to do this, by changing avro something like this in the `dependencies` block:

```
  shadow(library.java.avro) {
    exclude group: ....
  }
```

That makes clearer where the dependency is coming from. Though I guess it wouldn't prevent paranamer from coming in through other dependencies.